src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotAddEmailsToSamGroupStep.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.auth.iam.IamRole;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.List;
+import java.util.UUID;
+
+public class CreateSnapshotAddEmailsToSamGroupStep extends DefaultUndoStep {
```

Comment:
Is the idea here is that, on failure, the snapshot will be deleted so there's no need for an undo?

Reply:
Similar - but my thought was that the group would get deleted, so there isn't a big need to remove the users from the group that is already slated for deletion.